internal/handlers/pg/pgdb/settings.go

```diff
@@ -174,16 +173,44 @@ func getSettingsTable(ctx context.Context, tx pgx.Tx, db string) (*types.Documen
 	return settings, nil
 }
 
-// updateSettingsTable updates FerretDB settings table.
-func updateSettingsTable(ctx context.Context, tx pgx.Tx, db string, settings *types.Document) error {
+// setTableInSettings sets the table name for given collection in settings table.
+// As it's not possible to modify the settings table with a single operator (the data need to be retrieved first),
+// explicit lock is used to prevent concurrent modifications.
+// If the collection is already present in settings, ErrAlreadyExist will be returned.
+func setTableInSettings(ctx context.Context, tx pgx.Tx, db, collection, table string) error {
+	settings, err := getSettingsTable(ctx, tx, db, true)
+	if err != nil {
+		return lazyerrors.Error(err)
+	}
+
+	collections, ok := must.NotFail(settings.Get("collections")).(*types.Document)
+	if !ok {
+		return lazyerrors.Errorf("invalid settings document")
+	}
+
+	if collections.Has(collection) {
+		return ErrAlreadyExist
```

Comment:
That should be covered by tests

Reply:
I'm not sure it's possible to do it. 
I added a test that attempts to create a collection with the same name from a few parallel requests, but I'm not sure if it can reach this code as this case is very specific. 

Overall, the way how we work with settings needs to be simplified, and then we can reach better coverage. For example, I don't think we need that many requests of the settings table for every single operation, but this is how it's written, and it's not in the scope of the current task to refactor it (at least, this is not how the task is defined).

Do you have a particular example of a test that could be added?

Reply:
Let's panic then. We will know if that code can be reached